aptos-move/move-examples/token_objects/ambassador/client/ambassador.ts

```diff
@@ -0,0 +1,171 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import { AptosAccount, AptosClient, HexString } from "aptos";
+
+const NODE_URL = process.env.APTOS_NODE_URL || "https://fullnode.testnet.aptoslabs.com";
+const FAUCET_URL = process.env.APTOS_FAUCET_URL || "https://faucet.testnet.aptoslabs.com";
+
+function fetchGraphQL(
+    operationsDoc: string,
+    operationName: string,
+    variables: Record<string, any>
+) {
+    return fetch('https://indexer-testnet.staging.gcp.aptosdev.com/v1/graphql', {
+        method: 'POST',
+        body: JSON.stringify({
+            query: operationsDoc,
+            variables,
+            operationName,
+        }),
+    }).then(result => result.json());
+}
+
+async function getTokenAddr(ownerAddr: HexString, tokenName: string): Promise<HexString> {
+    const operation = `
+  query TokenOwnership {
+    current_token_ownerships_v2(
+      where: {owner_address: {_eq: "${ownerAddr}"}, amount: {_gt: 0}}
+      order_by: {last_transaction_version: desc}
+    ) {
+      amount
+      current_token_data {
+        token_uri
+        token_data_id
+        token_properties
+        token_name
+        current_collection {
+          collection_name
+        }
+      }
+      token_standard
+    }
+  }
+`;
+    const data = await fetchGraphQL(operation, "TokenOwnership", {});
```

Comment:
Is there a way to fetch the token's data (i.e. level and rank) using a single request/query? cc @0xmaayan @bowenyang007

Reply:
This single query also fetches the `rank` data. It is an entry of the property map, so `token_properties` should contain it. 
However, `level` is different because it's a top-level Move resource. Currently, a specific view function or a custom indexer can get the value. (@0xmaayan , @bowenyang007 )

Reply:
SDK provides view function call https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk/src/tests/e2e/aptos_client.test.ts#L602-L658
and Indexer has `move_resources` query

Reply:
Yeah I was thinking in the query you can join with move_resources to get level. This way you wouldn't need to send 2 separate requests to get all the data you need

Reply:
This example requests a separate query via calling the view function to get the AmbassadorLevel.

We could do it in a single query if we extended our custom indexer and interact with it: https://github.com/aptos-labs/aptos-indexer-processors/pull/24

Reply:
I see. May be you can leave that as a follow up for that example after the other examples are done